src/js/elements/mapchip.js

```diff
@@ -119,4 +158,53 @@ export class MapChip extends Observable {
         this.updateMapChipText(label);
         this.showMapChip(args);
     }
+
+    setAddFilterButton() {
+        if (this.filter.allDropdowns.length >= this.groups.length * 2) {
```

Comment:
Can you make this clearer - e.g.

let maxFilters = this.groups.length
let maxDropdowns = maxFilters * 2
let numDropdowns = this.filter.allDropdowns.length

if (numDropdowns >= maxDropdowns) {...}